### PR TITLE
OLD: first commit add navbar, add signout

### PR DESCRIPTION
**Pull request by RudyBecker**
- [x] add navbar
- [x] add sign-in/sign-out functionality to dropdown
- [x] hide sign in option in drop down when signed in (at XS and L widths)
- [ ] intermediary spinner on avatar photo when waiting for user avatar to load
- [x] ensure user remains logged in on refresh
- [ ] write tests

### CODE REVIEW

**Comment by RudyBecker** [comment]
https://user-images.githubusercontent.com/52896529/125853661-2f0f8af6-26a7-44dd-93bd-f14099e0b790.mov

File: tailwind.config.js
Comment (brandonaaskov) [inline tailwind.config.js:24]: All the changes in this file are forced style changes. My guess is that it's coming from either a setting in VS Code or a plugin. I personally don't care that much about style, which is why I specifically didn't implement a linter in this codebase: I like people to have the freedom to write what they want, plus it's like a mini fingerprint on who wrote it.

Generally speaking though, these kinds of changes are frowned upon since they're not related to the PR and they're style changes (ie not functional).

This pops up a handful of times in this PR. Can you revert changes to files/lines that didn't need to change for this PR? Totally fine if you want to do double quotes, semicolons, etc. but I just want to limit it to the code that was written.

File: src/store/auth.js
Comment (brandonaaskov) [inline src/store/auth.js:61]: Good name change 👍

File: src/store/auth.js
Comment (brandonaaskov) [inline src/store/auth.js:87]: This is duplicative of the `onAuthStateChanged` handler above. The handler in the store above will fire automatically when the app is "entered" (meaning any user loads their first page of the app in the browser). This method is actually setting up a new listener every time you call `authAction`, and we don't want that. We just want to setup that listener once. Since Firebase works entirely client-side, we have to wait for this response to come back before we can know if the user was previously logged in and their cookie hasn't expired.

Firebase has a method where you can just check the current user at any time, but instead of using that sporadically all over the app, it's easier  & cleaner to have a single listener in the auth store update its state with the user info instead.

TL;DR you should remove this method.

File: src/store/auth.js
Comment (brandonaaskov) [inline src/store/auth.js:79]: Huh... I can't believe I never had a method for this despite importing it on the index page.

It's not wrong, but it's also not pretty. We're not using the `.then` or `.catch` style promises in this app, and generally speaking I avoid them whenever possible especially now that [top-level await](https://v8.dev/features/top-level-await) is becoming more prevalent.

Also, I think this might be the first time I've ever seen someone use the second argument to `.then()` to catch the error. Like I said, it's not wrong, but that takes precedence over `.catch` and 99 times out of 100 you'll promises in this style written in the `.then().catch()` format. That said, that format is also ugly and async/await is best for readability and code flow.

Comment (brandonaaskov) [reply src/store/auth.js:79]: That reminds me, we actually don't care if this is successful: the onAuthStateChanged handler will update the store's state for us. But we do care if there was an error. So let's still log out that `console.error` you have, but can you change the second argument to it to be `error.message` instead? I typically find that's all I really care about in debugging and having the entire error object logged in the console is unnecessary.

**Review by brandonaaskov — CHANGES_REQUESTED** [review]
I left some comments requesting changes. I haven't gone through the markup yet. I realized that you opened this after forking it, but you have write access to this repo so can you create the PR in this repo instead of the forked branch?

File: src/components/NavBar.vue
Comment (brandonaaskov) [inline src/components/NavBar.vue:341]: This is the TailwindUI HTML template, but they offer a Vue template as well. We should leverage the Vue code they have instead since it's already broken into expected components.

File: src/pages/index.vue
Comment (brandonaaskov) [inline src/pages/index.vue:4]: It's very easy to camelcase all the things when living in javascript land. But `imgURL` here is an attribute on markup, and that convention falls kebab casing (e.g. `img-url`).

It's also good to think about naming things. What is this image for? Is it an icon? Is it a background image in the navbar? Is it the logo? It's much better to be specific when you can about what a variable is so that the brain has to do less thinking. I would also point out that that's not a URL, it's not even a path, it's just the filename. You handle the path in the Navbar component itself.

Think of every component as being as dumb as possible. The only logic a component should be doing is when it needs to figure out something based on its own state. In this case, the Navbar doesn't need to figure out the image path for the logo, we can just supply it directly:
```jsx
<NavBar logo-path="~/assets/images.rocket-insights-logo.svg"...
```

The `logo-path` attribute doesn't even have to be a variable, and in this case, it should just be a string since it's never going to change.

File: src/components/NavBar.vue
Comment (brandonaaskov) [inline src/components/NavBar.vue:9]: You definitely don't want to be importing the image this way. This is basically embedding the asset in the built nuxt app, making it much heavier than necessary. Images should always be loaded client-side, on demand as needed. This should be a path to the static asset. You're trying to load this in as a dynamic image, but that's definitely overkill for this.

**Comment by brandonaaskov** [comment]
I don't have this running locally, but in the preview deployment on Netlify I'm seeing this:
image

But using the markup directly from Tailwind, I get this:
image

So the styles are off for that. Also, even though Tailwind doesn't ship with javascript to handle the dynamic stuff like opening and closing that popover, we should add that functionality to the Navbar component so that clicking the avatar image opens and closes the popover.

File: tailwind.config.js
Comment (RudyBecker) [reply tailwind.config.js:24]: Created new PR, removed Prettier from workspace (which added all the formatting)

File: src/store/auth.js
Comment (RudyBecker) [reply src/store/auth.js:87]: Removed.

File: src/components/NavBar.vue
Comment (RudyBecker) [reply src/components/NavBar.vue:341]: Per Conversation, using HTML version as Tailwind Vue version has dependencies that do not work with Nuxt (Vue version 2.0) requires Vue 3.0.

File: src/store/auth.js
Comment (RudyBecker) [reply src/store/auth.js:79]: Updated with try/catch to accomodate being able to receive error.message if error on login. Upon researching top-level await didn't know if wanted to change auth.js store to auth.mjs file type or import "type":"module" across whole project in package.json. Happy to speak more about proper implementation. Found good resource here : https://www.stefanjudis.com/today-i-learned/top-level-await-is-available-in-node-js-modules/

File: src/components/NavBar.vue
Comment (RudyBecker) [reply src/components/NavBar.vue:9]: changed paths to static assets, removed dynamic.

**Pull request closed without merging**
